### PR TITLE
Upgrade to the most recent test package.

### PR DESCRIPTION
**Pull request by Hixie**
Also, add tests to verify that our coverage is actually being tested!

### CODE REVIEW

**Review by jason-simmons — APPROVED** [review]

File: packages/flutter_tools/test/commands/test_test.dart [hunk 1]
```diff
@@ -47,18 +47,33 @@ void main() {
...
+      if (!result.stdout.contains('+1: All tests passed'))
+        fail('unexpected output from test:\n\n${result.stdout}\n-- end stdout --\n\n');
+      expect(result.exitCode, 0);
+    });
+
+    testUsingContext('test runs to completion', () async {
+      Cache.flutterRoot = '../..';
+      final ProcessResult result = await _runFlutterTest('trivial', automatedTestsDirectory, flutterTestDirectory,
+        extraArgs: const <String>['--verbose']);
+      if ((!result.stdout.contains('+1: All tests passed')) ||
```
Comment (jacob314) [inline packages/flutter_tools/test/commands/test_test.dart]: nit: maybe make all these 1 regexp

**Review by jacob314 — APPROVED** [review]

**Comment by jacob314** [comment]
lgtm

File: packages/flutter_tools/test/commands/test_test.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/test/commands/test_test.dart]: Yeah, I considered that, but the pain of having a regular expression seemed to outweigh the likelihood of catching a bug where these end up out of order somehow.

**Comment by Hixie** [comment]
will land on green unless you feel strongly about the regexp thing

**Pull request merged**